plugins/kubernetes-backend/src/permissions/rules/isOfKind.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { RESOURCE_TYPE_KUBERNETES_RESOURCE } from '@backstage/plugin-kubernetes-common';
+import { createKubernetesPermissionRule } from './util';
+import { z } from 'zod';
+import { ObjectToFetch } from '../../types/types';
+import { PermissionCriteria } from '@backstage/plugin-permission-common';
+import { DEFAULT_OBJECTS } from '../../service/KubernetesFanOutHandler';
+
+/**
+ * A Resource {@link @backstage/plugin-permission-node#PermissionRule} which
+ * filters for resources by a given kind type.
+ *
+ * @alpha
+ */
+
+export const isOfKind = createKubernetesPermissionRule({
+  name: 'IS_OF_KIND',
+  description: 'Allow kubernetes resources with the specified kind',
+  resourceType: RESOURCE_TYPE_KUBERNETES_RESOURCE,
+  paramsSchema: z.object({
+    kind: z.string().describe('Kind to match'),
```

Comment:
Would an array helps here instead of a string or people will usually filter items using a single kind?

Reply:
Its my understanding that we want this rule to take a single kind parameter but that does not mean we can only filter by a single kind. Its my understanding that the conditions response itself returned to us by the permission api call can be composed/complex and that is what we actually use to filter data at our endpoint. So to satisfy a “i want to filter data by pods and deployments” scenario we can compose a complex condition with multiple isofkind calls via permission policies that would return something like  { anyOf : [ isOfKind({ kind: 'pods' }), isOfKind({ kind: 'deployments' }) ] } which  is a valid conditions for a decision response.